### PR TITLE
Fix docker builds

### PR DESCRIPTION
**Pull request by tonytheleg**
* Updates the Dockerfile to follow patterns used in other repos, plus it will ensure proper image managment by Boilerplate
* Enables FIPS in Makefile to ensure image is properly built to use FIPS
* Adds build scripts utilized by new Dockerfile
* includes changes from a boilerplate-update

### CODE REVIEW

File: build/Dockerfile [hunk 1]
```diff
@@ -1,35 +1,27 @@
 FROM quay.io/app-sre/boilerplate:image-v3.0.3 AS builder
-FROM golang:1.19 as builder
-ARG TARGETOS
-ARG TARGETARCH
 
-WORKDIR /workspace
-# Copy the Go Modules manifests
+RUN mkdir -p /workdir
```
Comment (mjlshen) [inline build/Dockerfile:3]:
```suggestion
```
WORKDIR will make a directory if it doesn't exist

File: build/bin/entrypoint [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#!/bin/sh -e
+
+# This is documented here:
+# https://docs.openshift.com/container-platform/4.10/openshift_images/create-images.html#images-create-guide-openshift_create-images
+
+if ! whoami &>/dev/null; then
+  if [ -w /etc/passwd ]; then
+    echo "${USER_NAME:-aws-vpce-operator}:x:$(id -u):$(id -g):${USER_NAME:-aws-vpce-operator} user:${HOME}:/sbin/nologin" >> /etc/passwd
```
Comment (mjlshen) [inline build/bin/entrypoint:8]:
```suggestion
    echo "${USER_NAME:-configure-goalert-operator}:x:$(id -u):$(id -g):${USER_NAME:-configure-goalert-operator} user:${HOME}:/sbin/nologin" >> /etc/passwd
```

**Review by mjlshen — CHANGES_REQUESTED** [review]

File: build/Dockerfile (hunk 1 above)
Comment (tonytheleg) [reply build/Dockerfile:3]: This is all your fault....

File: build/bin/entrypoint (hunk 2 above)
Comment (tonytheleg) [reply build/bin/entrypoint]: updated

**Pull request merged**